spring-web-modules/spring-mvc-basics-4/src/test/java/com/baeldung/optionalpathvars/ArticleViewerControllerIntegrationTest.java

```diff
@@ -1,54 +0,0 @@
-package com.baeldung.optionalpathvars;
-
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.test.context.web.WebAppConfiguration;
-import org.springframework.test.web.servlet.MockMvc;
-import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
-import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
-import org.springframework.test.web.servlet.setup.MockMvcBuilders;
-import org.springframework.web.context.WebApplicationContext;
-import com.baeldung.controller.config.WebConfig;
-
-@RunWith(SpringJUnit4ClassRunner.class)
-@WebAppConfiguration
-@ContextConfiguration(classes = { WebConfig.class })
-public class ArticleViewerControllerIntegrationTest {
```

Comment:
similar question here, do we need this test and the corresponding controller ArticleViewerController?

Reply:
Since I removed the controller I removed also the test.

Reply:
Can we rename this test to ArticleViewerWithRequiredAttributeControllerIntegrationTest since content of both the tests are same

Reply:
Ok, and then should we remove this one: ArticleViewerControllerWithRequiredAttributeIntegrationTest?

Reply:
I mean yeah, we can rename it to ArticleViewerWithRequiredAttributeControllerIntegrationTest

Reply:
I rename it into: ArticleViewerControllerWithRequiredAttributeIntegrationTest.